docs/connectors/connector-lifecycle.mdx

```diff
@@ -0,0 +1,195 @@
+sidebar_position: 6
+---
+
+:::info
+This document describes the lifecycle of a connector implemented in Go using the
+[Connector SDK](https://github.com/ConduitIO/conduit-connector-sdk), which
+abstracts away some parts of the Connector Protocol. If you are implementing a
+connector in a different language you should refer to the
+[Connector Protocol](https://github.com/ConduitIO/conduit-connector-protocol).
+:::
+
+Connector plugins, implemented using the
+[Connector SDK](https://github.com/ConduitIO/conduit-connector-sdk), can
+implement 2 interfaces: [`Source`](https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk#Source)
+for source connectors and [`Destination`](https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk#Destination)
+for destination connectors. Below we will describe the order of the method calls
+in sources and destinations, as well as when lifecycle events are triggered. For
+more information on how to implement a connector see the
+[Connector SDK godocs](https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk),
+the [Connector Template](https://github.com/ConduitIO/conduit-connector-template)
+or the [guide for building a connector](/guides/build-a-conduit-connector).
+
+## Lifecycle
+
+The lifecycle of the connector generally looks like this:
+
+1. The [connector plugin is loaded](#connector-plugin-is-loaded)
+2. The user [creates the connector](#connector-is-created)
+3. The user [starts and stops the pipeline](#pipeline-is-started-and-stopped)
+4. The user [deletes the connector](#connector-is-deleted)
+
+### Connector plugin is loaded
+
+Conduit loads all connector plugins it discovers in the
+[connector path](/docs/connectors/installing#installing-a-connector-in-conduit)
+during its start. Each plugin is initiated to ensure it is a valid binary and to
+obtain the connector specifications. These specifications are then stored in
+Conduit's memory and exposed through the API. This helps the UI display the
+available connector plugins and their corresponding parameters that can be
+configured.
+
+During the loading of connector specifications, the
+[`Source.Parameters`](https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk#Source)
+and
+[`Destination.Parameters`](https://pkg.go.dev/github.com/conduitio/conduit-connector-sdk#Destination)
+methods are invoked to retrieve the parameters. These methods are called only
+once, and no other methods are triggered before the connector process is
+stopped.
+
+### Connector is created
+
+Upon creating a connector ([`POST /v1/connectors`](/api#post-/v1/connectors)),
+Conduit validates the connector configuration. It starts an instance of the
+connector plugin and invokes the `Configure` method to validate the
+configuration. If the method returns no errors, the configuration is deemed
+valid, and the connector creation process continues. Otherwise, the creation of
+the connector is aborted. Afterward, Conduit signals the plugin to stop by
```

Comment:
This would make it seem that `Teardown` is only called if the connector creation is aborted, while it actually happens regardless of what `Configure` returns.

That said, I connected the previous two sentences to improve the flow.